ServiceDescriptor.cs

```diff
@@ -45,18 +45,24 @@ public ServiceDescriptor()
         {
             // find co-located configuration xml. We search up to the ancestor directories to simplify debugging,
             // as well as trimming off ".vshost" suffix (which is used during debugging)
+            //Get the first parent to go into the recursive loop
             string p = ExecutablePath;
             string baseName = Path.GetFileNameWithoutExtension(p);
             if (baseName.EndsWith(".vshost")) baseName = baseName.Substring(0, baseName.Length - 7);
+            DirectoryInfo d = new DirectoryInfo(Path.GetDirectoryName(p));
             while (true)
             {
-                p = Path.GetDirectoryName(p);
-                if (File.Exists(Path.Combine(p, baseName + ".xml")))
+                if (File.Exists(Path.Combine(d.FullName, baseName + ".xml")))
                     break;
+
+                if (d.Parent == null)
+                    throw new FileNotFoundException("Unable to locate "+baseName+".xml file within executable directory or any parents");

```

Comment:
This is using your same behavior but getting rid of your null issue.

The loop first checks for the file in directory.fullname with the basename, and if it doesnt exist, the directoryinfo object parent property returns null if you have searched up to C:\

See https://msdn.microsoft.com/en-us/library/system.io.directoryinfo.parent(v=vs.80).aspx

Basically your loop still redefines your string directory going from directory to directory.  But there is a potential you could search to c:\, still not find the xml file.  This is the root cause of your current null issue.  If you try to  find DirectoryName of C:\ you get null